### PR TITLE
Templates for extra links in top navbar and repo tool tabs.

### PR DESCRIPTION
**Pull request by AlbertoGP**
These templates add content to the top navbar and to the repository page.
This way we do not have to copy and modify the whole template, and re-modify it after upgrading Gitea if
it changes.

I'll put a comment with more details under "Custom tabs for projects in web ui #1199".

This fixes "Custom tabs for projects in web ui #1199", and the intention of "Custom Routing / new Pages?! #2115" (but without routing changes).

### CODE REVIEW

**Comment by lafriks** [comment]
Please place them under custom folder

**Comment by AlbertoGP** [comment]
Sure, no problem. Is it fine if the tmpl files do not exist by default?

Please note that both "extra_tabs.tmpl" files are just placeholders, with their content commented out so they have no effect. The real ones go indeed into "custom/templates/".

**Comment by lafriks** [comment]
I think at least empty files must exist

**Comment by AlbertoGP** [comment]
All right, I've made both `.tmpl` files empty.

**Review by techknowlogick — CHANGES_REQUESTED** [review]
@AlbertoGP I think @lafriks meant for you to put templates in the custom folder in the templates dir (so `custom/templates/custom`). The templates shouldn't go into `base` folder.

See #3051 as an example of how this has been done already. Also you can see #3286

**Comment by AlbertoGP** [comment]
Oh, I see now, thanks!
I'll put them in "templates/custom/", which also means they need different file names.

**Comment by lafriks** [comment]
Could you also add info about this in gitea customization docs?

**Comment by AlbertoGP** [comment]
Yes, of course: I tried doing that, but got blocked by "Documentation build: Partial "header.html" not found #3306".
I found a solution for that one, but to make it into a different pull request I need to move this one to another branch, which I'm doing right now.
Edit: no need to change this PR to another branch after all.

**Comment by thehowl** [comment]
CI fail seems to be unrelated. LGTM

**Review by sapk — APPROVED** [review]

File: docs/content/doc/advanced/customizing-gitea.en-us.md [hunk 1]
```diff
@@ -51,6 +51,8 @@ Any statement contained inside `{{` and `}}` are Gitea templete's syntax and sho
 
 To add in custom HTML to the header or the footer of the page, in the `templates/custom` directory there are `header.tmpl` and `footer.tmpl` that can be modified. This is useful if you want to add in custom CSS files, or additional Javascript.
 
+If all you want is to add extra links to the top navigation bar, or extra tabs to the repository view, you can put them in `templates/custom/extra_links.tmpl` and `templates/custom/extra_tabs.tmpl` inside your `custom/` directory.
```
Comment (lafriks) [inline docs/content/doc/advanced/customizing-gitea.en-us.md]: please change to:
you can put them in `extra_links.tmpl` and `extra_tabs.tmpl` inside your `templates/custom` directory

**Comment by lafriks** [comment]
@techknowlogick needs your approval

**Review by techknowlogick — APPROVED** [review]

**Comment by techknowlogick** [comment]
@lafriks thanks for the ping. approved.

**Pull request merged**